Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -718,6 +705,38 @@ theorem comap_inducing_of_surjective (hf : Surjective f) : Inducing (comap f) wh
     exact ⟨f '' F, hF.symm.trans (preimage_comap_zeroLocus f F)⟩
 #align prime_spectrum.comap_inducing_of_surjective PrimeSpectrum.comap_inducing_of_surjective
 
+
+end Comap
+end CommSemiRing
+
+section SpecOfSurjective
+
+/-! The comap of a surjective ring homomorphism is a closed embedding between the prime spectra. -/
+
+
+open Function RingHom
+
+variable [CommRing R] [CommRing S]
+variable (f : R →+* S)
+variable {R}
+
+theorem comap_singleton_isClosed_of_surjective (f : R →+* S) (hf : Function.Surjective f)
+    (x : PrimeSpectrum S) (hx : IsClosed ({x} : Set (PrimeSpectrum S))) :
+    IsClosed ({comap f x} : Set (PrimeSpectrum R)) :=
+  haveI : x.asIdeal.IsMaximal := (isClosed_singleton_iff_isMaximal x).1 hx
+  (isClosed_singleton_iff_isMaximal _).2 (Ideal.comap_isMaximal_of_surjective f hf)
```

Comment:
Once you generalize `Ideal.comap_isMaximal_of_surjective` you can probably generalize all results in this section that assumes surjectivity (not sure about IsIntegral). For surjectivity it should still be true for Semirings that there is an OrderIso between ideals in R containing the kernel and all ideals in S.